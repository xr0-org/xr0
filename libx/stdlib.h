#ifndef STDLIB_H
#define STDLIB_H

axiom void *
malloc(int size) [ .alloc result; ];

axiom void
free(void *ptr) [ .dealloc ptr; ];

#endif
