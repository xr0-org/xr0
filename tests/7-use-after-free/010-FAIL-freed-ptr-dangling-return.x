#include <stdlib.h>

int *
func();

int
main()
{
	int *i;
	int *j;

	i = func();
	j = *i;		/* ERROR: unjustified indirection */

	return 0;
}

int *
func()
{
	int *p;
	p = malloc(1);
	free(p);	/* p dangling */

	return p;	/* return dangling ptr */
}
