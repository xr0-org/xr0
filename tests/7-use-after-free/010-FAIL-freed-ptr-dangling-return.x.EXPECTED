7-use-after-free/010-FAIL-freed-ptr-dangling-return.x:13:8: cannot exec statement: unjustified indirection (rvalue)
